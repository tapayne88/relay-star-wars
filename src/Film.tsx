import React, { FC } from "react";
import graphql from "babel-plugin-relay/macro";
import { useFragment } from "react-relay/hooks";
import { Film_film$key } from "./__generated__/Film_film.graphql";

const Film: FC<Props> = ({ filmRef }) => {
  const { title, releaseDate, director } = useFragment(
    graphql`
      fragment Film_film on Film {
        title
        releaseDate
        director
      }
    `,
    filmRef
  );

  return (
    <div>
      {releaseDate} <strong>{title}</strong> ({director})
    </div>
  );
};

type Props = { filmRef: Film_film$key };

export default Film;
