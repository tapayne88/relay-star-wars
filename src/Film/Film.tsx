import React, { FC } from "react";
import graphql from "babel-plugin-relay/macro";
import { createFragmentContainer } from "react-relay";
import { Film_film } from "./__generated__/Film_film.graphql";

const Film: FC<Props> = ({ film: { title, releaseDate, director } }) => {
  return (
    <div>
      {releaseDate} <strong>{title}</strong> ({director})
    </div>
  );
};

type Props = { film: Film_film };

export default createFragmentContainer(Film, {
  film: graphql`
    fragment Film_film on Film {
      title
      releaseDate
      director
    }
  `,
});
