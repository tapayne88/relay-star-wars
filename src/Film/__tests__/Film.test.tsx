import React from "react";
import { QueryRenderer } from "react-relay";
import graphql from "babel-plugin-relay/macro";
import { createMockEnvironment, MockPayloadGenerator } from "relay-test-utils";
import ReactTestRenderer from "react-test-renderer";
import Film from "../Film";
import { Film_TestQuery } from "./__generated__/Film_TestQuery.graphql";

test("Fragment Container", () => {
  const environment = createMockEnvironment();
  const TestRenderer = () => (
    <QueryRenderer<Film_TestQuery>
      environment={environment}
      query={graphql`
        query Film_TestQuery @relay_test_operation {
          film: node(id: "test-id") {
            ...Film_film
          }
        }
      `}
      variables={{}}
      render={({ error, props }) => {
        if (props) {
          return <Film film={props.film!} />;
        } else if (error) {
          return error.message;
        }
        return "Loading...";
      }}
    />
  );
  const renderer = ReactTestRenderer.create(<TestRenderer />);
  environment.mock.resolveMostRecentOperation((operation) =>
    MockPayloadGenerator.generate(operation)
  );

  expect(renderer).toMatchSnapshot();
});
