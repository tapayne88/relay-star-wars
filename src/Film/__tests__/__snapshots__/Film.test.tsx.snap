// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fragment Container 1`] = `
<div>
  &lt;mock-value-for-field-"releaseDate"&gt;
   
  <strong>
    &lt;mock-value-for-field-"title"&gt;
  </strong>
</div>
`;
